docs/explanation/parts.rst

```diff
@@ -0,0 +1,263 @@
+Parts
+=====
+
+In the Craft Parts framework, parts are descriptions of the components to be
+build for deployment, either individually or as part of a larger build
+containing many components.
+
+Tools like `Snapcraft`_ and `Juju`_ that use the concepts of parts to
+describe a build process typically accept specifications of parts in YAML format. This allows each part to be described in a convenient,
+mostly-declarative format.
+
+When the Craft Parts framework is used to process a part on behalf of a tool
+or library, it performs the steps described in the
+:ref:`parts lifecycle <lifecycle>` to *pull* the source code and dependencies, *overlay* them onto a base file system, *build* the part, then *organize*,
+*stage* and *prime* the build products for further processing.
```

Comment:
note: It may be too much detail for an overview, but applications can opt out of following these steps (i.e. charmcraft doesn't use overlays)

Reply:
This is probably better detail for `lifecycle.rst`, which currently doesn't say anything about them being optional. (That's my fault.)

Reply:
It's OK. I wanted it to be complete, at least to start with. I will start to move the finer points out into other documents if they distract from the explanation. I could add something to the lifecycle about step-skipping, perhaps.